x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/jwt/JwtRealmSettings.java

```diff
@@ -284,6 +266,61 @@ private JwtRealmSettings() {}
         key -> Setting.intSetting(key, DEFAULT_HTTP_MAX_ENDPOINT_CONNECTIONS, MIN_HTTP_MAX_ENDPOINT_CONNECTIONS, Setting.Property.NodeScope)
     );
 
+    // Individual outgoing HTTP proxy settings
+
+    public static final Setting.AffixSetting<String> HTTP_PROXY_SCHEME = Setting.affixKeySetting(
```

Comment:
For JWT validation, I copied HTTP Client code from OIDC realm. That code already handled HTTP Proxy settings, it seemed like more work to remove it, so I left it in for discussion here in PR review.

I can add HTTP option for Proxy scheme. My assumption was if JWKSet was HTTPS, then the proxy needed to be HTTPS too. I left in with HTTPS only option, so we could discuss it here in PR review.

Reply:
In general, I'm of the option of not putting unused code in master when we can and thus I suggest either adding full support for proxy here or remove references and add proxy support in a follow up PR ( I'm in favor of the latter ) 

No matter what we do, I believe we should support both http and https for the proxy for the reasons stated above.

Reply:
I agree with supporting http and https.

Reply:
I get what you are saying. I looked at it another way, though.

The original plan was add all the settings in the original PR, but don't use them. In subsequence PRs, we would use them. That is why HTTP settings were included in the settings, but still not used in the realm yet.
```
testHttpProxySchemeSetting()
testHttpProxyPortSetting()
testHttpProxyHostSetting()
```

When I reused the JWT validation code from OIDC, I realized there was minimal effort to reuse the HTTP proxy settings. So, they are grouped together with the other HTTP and TLS settings, which are all not being used yet. When time comes to work on the HTTP and TLS realm code, the HTTP settings with HTTP proxy settings are all in place. It would have been more work to remove the HTTP proxy settings, and add them back later. They will end up getting used with the HTTP settings, and tested together, so the end result should be no unused code at the end of the initial release.

Let me know what you think, if it still makes sense to remove the HTTP proxy code. I think if we remove it, it will end up getting deferred. If we leave it in, I think it should work because we know the code reused from OIDC is working.

Reply:
The unused code in settings was a by-product of wanting to split the PRs in a way so that they don't get too big which is slightly different in my view. The entire PR was "unused code". Now we're adding part of a feature because it was easier to copy existing code. 

>  I think if we remove it, it will end up getting deferred. 

That's always up to us to not happen. I don't think we need to partially add code in order to keep us on target.

This is mostly for the sake of discussion, I won't block on this.

Reply:
Proxy removed.